From 5367abf84e094876c27724b19e1b5b496763fdca Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:04 -0400
Subject: [acpi] hotplug/pci: Drop acpiphp_bus_add()

Message-id: <1411129776-21656-30-git-send-email-prarit@redhat.com>
Patchwork-id: 96366
O-Subject: [RHEL7.1 PATCH BZ 1141846 29/61] ACPI / hotplug / PCI: Drop acpiphp_bus_add()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit b6708fbf98ac01d27c8d4d7f7b4fa87583b658cc
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Tue Feb 4 00:39:58 2014 +0100

    ACPI / hotplug / PCI: Drop acpiphp_bus_add()

    acpiphp_bus_add() is only called from one place, so move the code out
    of it into that place and drop it.  Also make that code use
    func_to_acpi_device() to get the struct acpi_device pointer it needs
    instead of calling acpi_bus_get_device() which may be costly.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Tested-by: Mika Westerberg <mika.westerberg@linux.intel.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index d3634fb..36e0593 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -491,20 +491,6 @@ static unsigned char acpiphp_max_busnr(struct pci_bus *bus)
  return max;
 }
 
-/**
- * acpiphp_bus_add - Scan ACPI namespace subtree.
- * @handle: ACPI object handle to start the scan from.
- */
-static void acpiphp_bus_add(acpi_handle handle)
-{
- struct acpi_device *adev = NULL;
-
- acpi_bus_scan(handle);
- acpi_bus_get_device(handle, &adev);
- if (acpi_device_enumerated(adev))
-  acpi_device_set_power(adev, ACPI_STATE_D0);
-}
-
 static void acpiphp_set_acpi_region(struct acpiphp_slot *slot)
 {
  struct acpiphp_func *func;
@@ -544,9 +530,13 @@ static int acpiphp_rescan_slot(struct acpiphp_slot *slot)
 {
  struct acpiphp_func *func;
 
- list_for_each_entry(func, &slot->funcs, sibling)
-  acpiphp_bus_add(func_to_handle(func));
+ list_for_each_entry(func, &slot->funcs, sibling) {
+  struct acpi_device *adev = func_to_acpi_device(func);
 
+  acpi_bus_scan(adev->handle);
+  if (acpi_device_enumerated(adev))
+   acpi_device_set_power(adev, ACPI_STATE_D0);
+ }
  return pci_scan_slot(slot->bus, PCI_DEVFN(slot->device, 0));
 }
 
-- 
1.7.1